public/components/common/hooks/use-filter-manager.ts

```diff
@@ -10,10 +10,24 @@
  * Find more information about this on the LICENSE file.
  */
 import { getDataPlugin } from '../../../kibana-services';
-import { useState, useEffect} from 'react';
-
+import { useState, useEffect, useMemo } from 'react';
+import { Filter } from 'src/plugins/data/public';
+import _ from 'lodash';
 
 export const useFilterManager = () => {
-    const [filterManager, setFilterManager] = useState(getDataPlugin().query.filterManager);
-    return filterManager;
-}
\ No newline at end of file
+  const filterManager = useMemo(() => getDataPlugin().query.filterManager, []);
+  const [filters, setFilters] = useState<Filter[]>([])
```

Comment:
**thoughts**: Why the initial value of `filters` state is an empty array `[]`. Should we use the `filterManager.getFilters()` as the initial value?

Reply:
Since the call for filters is asynchronous we couldn't use it as an innitial value. We could create an explicit `useEffect` that calls for the filters to set them, but in my testing this doesn't seem to be necessary.